src/coreclr/src/vm/frames.cpp

```diff
@@ -480,6 +480,29 @@ void Frame::PopIfChained()
 
 //-----------------------------------------------------------------------
 #endif // #ifndef DACCESS_COMPILE
+
+//---------------------------------------------------------------
+// Whenever signature parsing is required, we need to make sure to
+// use the canonical function (if applicable) instead of the instantiated function. Failure
+// to do so may result in calls into the type loader to construct new types, which can be forbidden
+// in some situations (ex: during GC-related stack walking).
+//---------------------------------------------------------------
+MethodDesc* TransitionFrame::GetCanonicalFunction()
+{
+    MethodDesc* pFunction = GetFunction();
+
+#ifndef DACCESS_COMPILE
+    if (pFunction != NULL && pFunction->IsInstantiatingStub())
+    {
+        // We need to work with the canonical method to avoid calling into the type loader to
+        // construct new types
+        pFunction = pFunction->GetWrappedMethodDesc();
```

Comment:
The wrapped method desc may have different signature than the instantiating stub. Is it always going to walk the arguments correctly?

Reply:
I see your point. Just to confirm, this is because of the addition of the extra instantiation argument, which can shift things, right?

Reply:
Yep, looks like the mismatch would cause GC holes if we use the wrapped method desc for signature parsing...